src/v/storage/segment_appender.cc

```diff
@@ -594,6 +633,44 @@ ss::future<> segment_appender::hard_flush() {
       });
 }
 
+bool segment_appender::inflight_write::try_merge(
+  const inflight_write& other, size_t pco) {
+    if (state == QUEUED && chunk == other.chunk) {
```

Comment:
not quite seeing where chunk and other.chunk aren't equal like when inflight.back() doesn't contain _head as the chunk, i think?

Reply:
Are you asking when in practice the `chunk == other.chunk` condition would be false?

We switch to a new chunk every time the current chunk is full, so the first write to the new chunk will generally fail this condition since the last write (`_inflight.back()`) is using the old chunk and the current write will be using a new chunk.

I don't think there's any other case since `_inflight` writes are popped from the other end, so when we are still within a chunk this will always be true if the `_inflight` is not empty.

Reply:
This condition is of course super important since merging writes from different chunks doesn't make any sense.

Reply:
>Are you asking when in practice the chunk == other.chunk condition would be false?

yeh. i think i see it now. so other.chunk is always whatever _head is (it's set in the chunk right before try_merge is called). and then _head may be different than _inflight.back().

Reply:
> yeh. i think i see it now. so other.chunk is always whatever _head is (it's set in the chunk right before try_merge is called). and then _head may be different than _inflight.back().

That's correct. The way we call right side is always the "current chunk" and the left side (`this`) is always the "chunk from the previous write", though we don't call this _every_ write since sometimes the last write will have already completed and then `_inflight` will be empty.